zenith_utils/src/sync.rs

```diff
@@ -0,0 +1,174 @@
+use pin_project_lite::pin_project;
+use std::future::Future;
+use std::marker::PhantomData;
+use std::pin::Pin;
+use std::{io, task};
+
+pin_project! {
+    /// We use this future to mark certain methods
+    /// as callable in both sync and async modes.
+    #[repr(transparent)]
+    pub struct SyncFuture<S, T: Future> {
+        #[pin]
+        nested: T,
```

Comment:
I'd rename this to `inner`, since it's [somewhat](https://doc.rust-lang.org/src/std/fs.rs.html#92-94) of a convention. This is a tiny nitpick and it doesn't matter that much, but this is a difficult file to understand so anything helps.

Reply:
No problem!